crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -35,16 +35,25 @@ impl<'db> Unpacker<'db> {
         self.context.db()
     }
 
-    /// Unpack the value to the target expression.
     pub(crate) fn unpack(&mut self, target: &ast::Expr, value: UnpackValue<'db>) {
         debug_assert!(
             matches!(target, ast::Expr::List(_) | ast::Expr::Tuple(_)),
-            "Unpacking target must be a list or tuple expression"
+            "Unpacking target must be a list or tuple"
         );
-
         let mut value_ty = infer_expression_types(self.db(), value.expression())
             .expression_ty(value.scoped_expression_id(self.db(), self.scope));
 
+        let is_in_stub_file = value
+            .expression()
+            .file(self.db())
+            .is_stub(self.db().upcast());
```

Comment:
This can now use `InferContext::in_stub` method, right?

Reply:
Oops, some simultaneous reviewing going on -- see my suggestion in https://github.com/astral-sh/ruff/pull/14982/files#r1903322714